src/users/dashboards.rs

```diff
@@ -173,20 +173,29 @@ impl Dashboards {
         }
 
         let mut s = self.0.write().expect(LOCK_EXPECT);
-        s.append(&mut this);
+        s.extend(
+            this.into_iter()
+                .map(|d| (d.dashboard_id.clone().unwrap_or_else(|| d.name.clone()), d)),
+        );
```

Comment:
Could we do this using `HashMap::insert`s, with `dashboard_id` as always the key!